lib/widgets/sort_posts_picker.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:lemmy_api_client/lemmy_api_client.dart';
+
+import 'bottom_modal.dart';
+
+class SortPostsPicker extends HookWidget {
+  final Function(SortType sort) onChange;
+  final SortType defaultSort;
+
+  SortPostsPicker({
+    @required this.onChange,
+    this.defaultSort = SortType.active,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    var sort = useState(defaultSort);
+
+    void selectSortType(BuildContext context) {
+      showModalBottomSheet(
+        isScrollControlled: true,
+        context: context,
+        backgroundColor: Colors.transparent,
+        builder: (context) => BottomModal(
+            title: 'sort by',
+            child: Column(
+              children: [
+                for (var x in SortType.values)
+                  RadioListTile<SortType>(
+                    value: x,
+                    groupValue: sort.value,
+                    title: Text(x.toString()),
```

Comment:
Use `describeEnum` provided by flutter, or add a TODO that human strings need to be added